test/unit/org/apache/cassandra/cql3/validation/entities/SecondaryIndexTest.java

```diff
@@ -1052,6 +1055,104 @@ public void testIndexQueriesWithIndexNotReady() throws Throwable
             execute("DROP index " + KEYSPACE + ".testIndex");
         }
     }
+    
+    @Test // A Bad init could leave an index only accepting reads
+    public void testReadOnlyIndex() throws Throwable
+    {
+        LoadTypeConstrainedIndex index = null;
+        try
+        {
+            // Inserts don't go through
+            LoadTypeConstrainedIndex.setSupportedLoad(LoadType.READ);
+            String tableName = createTable("CREATE TABLE %s (pk int, ck int, value int, PRIMARY KEY (pk, ck))");
+            String indexName = createIndex("CREATE CUSTOM INDEX ON %s (value) USING '" + LoadTypeConstrainedIndex.class.getName() + "'");
+            assertTrue(waitForIndex(keyspace(), tableName, indexName));
+            execute("SELECT value FROM %s WHERE value = 1");
+            execute("INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 1, 1, 1);
+            index = (LoadTypeConstrainedIndex) getCurrentColumnFamilyStore().indexManager.getIndexByName(indexName);
+            assertEquals(0, index.rowsInserted.size());
+
+            // Upon recovery, we can index data again
+            index.reset();
+            getCurrentColumnFamilyStore().indexManager.rebuildIndexesBlocking(ImmutableSet.of(indexName));
+            execute("INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 2, 1, 1);
+            assertEquals(1, index.rowsInserted.size());
+            execute("DROP index " + KEYSPACE + "." + indexName);
```

Comment:
Nit: we can use the provided `dropIndex` method:
```suggestion
            dropIndex("DROP INDEX %s." + indexName);
```
There are three more usages below.